### ISSUE TITLE
Problem：converting the data format to kitti

### ISSUE THREAD

**Issue by deyang2000** [AIR-THU/DAIR-RCooper#9]
Hi there!
Thanks for your excellent work.
 I intend to train and test Bevheight on your RCooper dataset (https://github.com/ADLab-AutoDrive/BEVHeight).
I noticed that you have provided the RCooper to DAIR-V2X conversion function. I have completed this step and converted the data set to DAIR-V2X format.
In the Bevheight program, the kitti format must be used for training, and Bevheight provides a function to convert from dair to kitti. （https://github.com/ADLab-AutoDrive/BEVHeight/blob/main/docs/prepare_dataset.md）.

I followed their guidebook and linked the RCooper dataset (after Data Conversion) with /data/dair-v2x-i (I didn't change the name to minimize code changes).

The data set I'm using here is Rcooper/converted_DAIR_dataset/intersection/infrastructure-side. Then, the following error occurred.
![image](https://github.com/AIR-THU/DAIR-RCooper/assets/129513825/e9d5185c-84f0-4329-807f-5841b91cbefe)

I tried to solve this problem by:
1. I noticed that Rcooper/converted_DAIR_dataset/corridor/infrastructure-side/label  only had virtuallidar , so I copied it and renamed it "camera" which existed in dair-v2x , however, it did not work.
2. I tried to delete the first few items in the label/virtuallidar  (000000-000007.json), but that didn't work either.

I am confident that my environment is pleasing (at least for Bevheight), and I have successfully trained and tested Bevheight on DAIR-V2X-I.

Do you know how I can solve this problem? I am still looking for a way. A smooth conversion of RCooper to kitti format is feasible and essential, and I hope someone can help me out.

**Comment by ryhnhao** [comment]
Sorry for the inconvenience.

We tested the RCooper -> DAIR pipeline for employing the tracking part in DAIR-V2X toolkit. It works in our attempts now.

Actually, to run the tracking part in DAIR-V2X, we also run the DAIR2KITTI, you may refer to [this part](https://github.com/AIR-THU/DAIR-RCooper/blob/main/codes/ab3dmot_plugin/data_convert/coop_label_dair2kitti.py) for help (Although it's a code for converting cooperative label, hope this can help you).

Indeed, the conversion of single-side label to kitti format is essential. But we are not sure if we might find time to get to it. Do feel free to submit a PR (RCooper/codes/ab3dmot_plugin/data_convert/single_label_dair2kitti.py) if you plan to implement this.

**Comment by deyang2000** [comment]
Thank you for your reply! I will refer to your suggestions for further improvement.

**Comment by deyang2000** [comment]
[README for Tracking](https://github.com/AIR-THU/DAIR-RCooper/blob/e54b446697e5f20e39f38a89ee4ca6af33f8585a/docs/tracking.md)

This file is not in the program：pip install -r code/ab3dmot_plugin/requirements.txt

Is this file the same as the requirements.txt in origin ab3dmot ?

wheel==0.37.1
scikit-learn==0.19.2
filterpy==1.4.5
numba==0.43.1	
matplotlib==2.2.3
pillow>=8.3.2
opencv-python==4.2.0.32
glob2==0.6
PyYAML==5.4
easydict==1.9
llvmlite==0.32.1 

By the way, this problem also happened in v2v4real_plugin part. Please check whether related files exist.

**Comment by deyang2000** [comment]
Hello there!
I want to use the 2d_box parameter.
 I noticed that:

![image](https://github.com/AIR-THU/DAIR-RCooper/assets/129513825/5c08efbd-82f5-44b4-ac6b-f5412e0f6572)

                with open(os.path.join(self.src_path_label, scene_id, agent_id, seq_id, filename), 'r') as f_label:
                    annos = json.load(f_label)
                print(annos)

Then I got something like  : 
_{'type': 'pedestrian', 'occluded_state': 0, 'truncated_state': 0, 'crowding': 0, 'ignore': 0, 'track_id': 0, '3d_location': {'x': 6.352496039733496, 'y': -18.776476275807994, 'z': -5.2088761335874105}, '3d_dimensions': {'w': 0.5230683244822689, 'h': 1.6016319666737393, 'l': 0.6154060885195027}, 'rotation': -1.436087041972111}_
 
That is, there is no 2d_box parameter.
Is there a 2d_box in the original data, and if not, what relations can I use to calculate the 2d_box? 

Thank you for getting back to me.

**Comment by deyang2000** [comment]
@ryhnhao Thanks for the trouble.

**Comment by deyang2000** [comment]
Good. I finally got this all worked out. I will train with BEVHeight on Rcooper and then compare it with the results in the paper.

**Comment by ryhnhao** [comment]
Sorry for the late reply.

Since the original annotation process is conducted from the LiDAR view, the 2D boxes infos are ignored. 

[mmdet3d](https://github.com/open-mmlab/mmdetection3d/blob/main/tools/dataset_converters/kitti_converter.py) may provide a possible solution for this.
